repository/src/main/java/org/apache/atlas/repository/store/graph/v2/UniqueQNAttributeMigrationService.java

```diff
@@ -0,0 +1,92 @@
+package org.apache.atlas.repository.store.graph.v2;
+
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.atlas.repository.graph.GraphHelper;
+import org.apache.atlas.repository.graphdb.AtlasVertex;
+import org.apache.commons.collections.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+public class UniqueQNAttributeMigrationService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(UniqueQNAttributeMigrationService.class);
+
+    private final EntityGraphRetriever entityRetriever;
+
+
+    private Set<String> entityGuids;
+    private final TransactionInterceptHelper   transactionInterceptHelper;
+
+    private final String QUALIFIED_NAME_ATTR = "qualifiedName";
+    private final String UNIQUE_QUALIFIED_NAME_ATTR = "__u_qualifiedName";
+
+    public UniqueQNAttributeMigrationService(EntityGraphRetriever entityRetriever, Set<String> entityGuids, TransactionInterceptHelper transactionInterceptHelper) {
+        this.entityRetriever = entityRetriever;
+        this.transactionInterceptHelper = transactionInterceptHelper;
+        this.entityGuids = entityGuids;
+    }
+
+    public void migrateQN() throws Exception {
+        try {
+            int count = 0;
+            for (String entityGuid : entityGuids) {
+                AtlasVertex entityVertex = entityRetriever.getEntityVertex(entityGuid);
+
+                if (entityVertex == null) {
+                    LOG.error("Entity vertex not found for guid: {}", entityGuid);
+                    continue;
+                }
+
+                boolean isCommitRequired = migrateuniqueQnAttr(entityVertex);
+                if (isCommitRequired){
+                    count++;
+                }
+                else {
+                    LOG.info("No changes to commit for entity: {} as no migration needed", entityGuid);
+                }
+            }
+
+            if (count > 0) {
+                LOG.info("Total Vertex updated: {}", count);
+                commitChanges();
+            }
+            else {
+                LOG.info("No changes to commit for entities as no migration needed");
+            }
```

Comment:
This logic still does not control commits properly
In case the client passes huge entityGuids, the commit might fail in that cases.

Since we are designing API from bulk POV, we must have guardrails for commit size, e.g. commit only if 
* count == threshold (ideal 20)
* count > 0 && no more entityGuids left to process

Reply:
I would suggest forget about the migration WF/Script
Only think about the API to repair unique QN which should support multiple guids at a time
Once we handle this properly, WF/Script can use it as needed (single guid at a time OR multiple at a time)